library/fsio.c

```diff
@@ -0,0 +1,395 @@
+/*
+ *  Serialization based implementation of filesystem IO functions.
+ *
+ *  Copyright (C) 2017, ARM Limited, All Rights Reserved
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may
+ *  not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  This file is part of mbed TLS (https://tls.mbed.org)
+ */
+
+#if !defined(MBEDTLS_CONFIG_FILE)
+#include "mbedtls/config.h"
+#else
+#include MBEDTLS_CONFIG_FILE
+#endif
+
+#if defined(MBEDTLS_FS_IO)
+
+#include <string.h>
+#include <assert.h>
+#include "mbedtls/fsio.h"
+
+#if !defined(MBEDTLS_FS_IO_ALT)
+
+#if !defined(MBEDTLS_PLATFORM_NO_STD_FUNCTIONS)
+
```

Comment:
If `MBEDTLS_FS_IO` is defined but both `MBEDTLS_FS_IO_ALT` and `MBEDTLS_PLATFORM_NO_STD_FUNCTIONS` are undefined, we're not emitting any definition. Should we document resp. check for that dependency in `config.h` resp. `check_config.h`?

Reply:
Good point. I planned to add this but forgot.